internal/devbox/util.go

```diff
@@ -5,73 +5,70 @@ package devbox
 
 import (
 	"context"
+	"io"
 	"io/fs"
 	"os"
 	"path/filepath"
 
 	"github.com/pkg/errors"
-	"go.jetpack.io/devbox/internal/devpkg"
-	"go.jetpack.io/devbox/internal/nix"
 
+	"go.jetpack.io/devbox/internal/devbox/devopt"
 	"go.jetpack.io/devbox/internal/xdg"
 )
 
-// addDevboxUtilityPackage adds a package to the devbox utility profile.
-// It's used to install applications devbox might need, like process-compose
-// This is an alternative to a global install which would modify a user's
-// environment.
-func (d *Devbox) addDevboxUtilityPackage(ctx context.Context, pkgName string) error {
-	pkg := devpkg.PackageFromStringWithDefaults(pkgName, d.lockfile)
-	installables, err := pkg.Installables()
+const processComposeVersion = "1.5.0"
+
+var utilProjectConfigPath string
+
+func initDevboxUtilityProject(ctx context.Context, stderr io.Writer) error {
+	devboxUtilityProjectPath, err := ensureDevboxUtilityConfig()
 	if err != nil {
 		return err
 	}
-	profilePath, err := utilityNixProfilePath()
+
+	box, err := Open(&devopt.Opts{
+		Dir:    devboxUtilityProjectPath,
+		Stderr: stderr,
+	})
 	if err != nil {
-		return err
+		return errors.WithStack(err)
 	}
 
-	for _, installable := range installables {
+	// Add all utilities here.
+	utilities := []string{
+		"process-compose@" + processComposeVersion,
+	}
```

Comment:
I think there's a chance that in the future we may want to install as needed. But since we only have one, this if fine.